server/src/actors/events.rs

```diff
@@ -125,6 +142,162 @@ impl EventProcessor {
+            .rev()
+            .find_map(|breadcrumb| breadcrumb.value().and_then(|b| b.timestamp.value()));
+
+        if timestamp1 > timestamp2 {
+            std::mem::swap(&mut breadcrumbs1, &mut breadcrumbs2);
+        }
+
+        let max_length = std::cmp::max(breadcrumbs1.len(), breadcrumbs2.len());
+
+        breadcrumbs1.extend(breadcrumbs2);
+
+        if breadcrumbs1.len() > max_length {
+            // keep only the last max_length elements from the vectors
+            breadcrumbs1.drain(0..(breadcrumbs1.len() - max_length));
+        }
+
+        if let Some(evt) = evt.value_mut() {
+            evt.breadcrumbs = Annotated::new(Values {
+                values: Annotated::new(breadcrumbs1.into()),
+                other: Object::default(),
+            });
+        }
+    }
+
+    /// Writes a placeholder to indicate that this event has an associated minidump.
+    fn write_minidump_placeholder(evt: &mut Annotated<Event>) {
+        let evt = evt.get_or_insert_with(Event::default);
+
+        let platform = evt.platform.value_mut();
+        *platform = Some("native".to_string());
+
+        let level = evt.level.value_mut();
+        *level = Some(Level::Fatal);
+
+        let exceptions = evt
+            .exceptions
+            .value_mut()
+            .get_or_insert_with(Values::<Exception>::default)
+            .values
+            .value_mut()
+            .get_or_insert_with(|| vec![]);
+
+        exceptions.clear(); // clear previous errors if any
+
+        exceptions.push(Annotated::new(Exception {
+            ty: Annotated::new("Minidump".to_string()),
+            value: Annotated::new(JsonLenientString("Invalid Minidump".to_string())),
+            mechanism: Annotated::new(Mechanism {
+                ty: Annotated::from("minidump".to_string()),
+                handled: Annotated::from(false),
+                synthetic: Annotated::from(true),
+                ..Mechanism::default()
+            }),
+            ..Exception::default()
+        }));
+    }
+
+    /// merge form data (from a multipart request into the event
+    fn event_from_minidump_data(
+        form_data: Item,
+        event_data_mps: Option<Item>,
+        breadcrumbs1: Option<Item>,
+        breadcrumbs2: Option<Item>,
+    ) -> Annotated<Event> {
+        // if we have a "sentry" entry it should be a straight json.. merge this into
+        if form_data.ty() != ItemType::FormData
+            || form_data.content_type() != Some(&ContentType::Text)
+        {
+            //this should never happen caller called it with
+            log::error!("merge_form_data called without a proper FormData item");
+            return Annotated::new(Event::default());
+        }
+
+        #[derive(serde::Deserialize)]
+        struct StringPair<'a>(pub &'a str, pub &'a str);
+
+        let payload = form_data.payload();
+        let mut consolidated_data_stream =
+            serde_json::Deserializer::from_slice(payload.deref()).into_iter();
```

Comment:
It's possible to declare the type here. This might help to make the type here a little more explicit and you can elide types down below.

```suggestion
            serde_json::Deserializer::from_slice(payload.deref()).into_iter::<StringPair>();
```